Settings.c

```diff
@@ -421,6 +427,7 @@ Settings* Settings_new(int cpuCount) {
       CRT_restorePrivileges();
    }
    this->colorScheme = 0;
+   this->debuggerTool = xStrdup("gdb");
```

Comment:
Maybe htop should check the ownership of the configuration file it's about to read, so that no un-privileged user can by accident modify a privileged configuration file and inject commands.

(Another unrelated personal thought is to set the value to NULL, i.e. disable the feature by default, so if someone fat-fingers the key `d` nothing happens.)

Reply:
Should I add an ownership check in general and refuse to read the file if (g+w || o+w) is set, or just for this specific setting?

Reply:
Re your unrelated personal thought: I don't think that's necessary; if people don't have `gdb` installed, nothing will happen when they fat-finger `d`. If they *do* have `gdb` installed, well, it's likely that they know what they're doing, and if they pressed the key by accident they know how to leave `gdb`.

Reply:
Re again: I've changed it to `D` which should be less likely to be hit by accident.

Reply:
Regarding your request to check the ownership:

I think that this could be accomplished using the following code:

```c
   bool trusted = true;
   struct stat64 file_stat;
   fstat64(fileno(fd), &file_stat);
   if (file_stat.st_uid != getuid32()) {
      printf("warning: htop settings file is not owned by uid %d\n", getuid32());
      trusted = false;
   } else if (file_stat.st_gid != getgid32()) {
      printf("warning: htop settings file is not owned by gid %d\n", getgid32());
      trusted = false;
   } else if (file_stat.st_mode & 0002) {
      printf("warning: htop settings file is writable by others\n");
      trusted = false;
   }
```

And then ignoring certain values if `trusted == false`.

However, having implemented this, I strongly oppose the idea because:

- It can be easily circumvented by an attacker who has write permissions to any folder in the hierarchy
- It prevents the config from working when I launch htop with `sudo`
- If somebody manages to manipulate the `htop` config in my home folder, they could just as well inject stuff into my `.bashrc`. Other tools like `bash` don't check their config files for permissions either.
- `ssh` used to check its config files for permission bits but this was removed because it caused too much pain and actually decreased security by e.g. preventing users from putting their secret keys onto flash drives
- I think ACLs would be ignored